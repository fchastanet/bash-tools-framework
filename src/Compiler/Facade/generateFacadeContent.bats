#!/usr/bin/env bash

# shellcheck source=src/batsHeaders.sh
source "$(cd "${BATS_TEST_DIRNAME}/../.." && pwd)/batsHeaders.sh"

# shellcheck source=/src/Compiler/Facade/generateFacadeContent.sh
source "${srcDir}/Compiler/Facade/generateFacadeContent.sh"
# shellcheck source=/src/Filters/directive.sh
source "${srcDir}/Filters/directive.sh"

function Compiler::Facade::generateFacadeContent::script2Facades { #@test
  run Compiler::Facade::generateFacadeContent "${BATS_TEST_DIRNAME}/testsData/script2Facades.sh"
  assert_success
  assert_output ""
}

function Compiler::Facade::generateFacadeContent::longerScript { #@test
  run Compiler::Facade::generateFacadeContent "${BATS_TEST_DIRNAME}/testsData/longerScript.sh"
  assert_success
  assert_output "$(cat "${BATS_TEST_DIRNAME}/testsData/generateFacadeContent.longerScript.expected.txt")"
}
