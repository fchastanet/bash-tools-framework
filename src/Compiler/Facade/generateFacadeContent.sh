#!/usr/bin/env bash

# @description generate facade content from original script file
# @arg $1 scriptFile:String the file from which a facade will be generated
# @exitcode 1 on script file not found or non readable file
# @stderr diagnostics information is displayed
Compiler::Facade::generateFacadeContent() {
  local scriptFile="$1"

  Filters::directive "${FILTER_DIRECTIVE_REMOVE_HEADERS}" "${scriptFile}" | sed -E '{/^#!/d;}'
}
