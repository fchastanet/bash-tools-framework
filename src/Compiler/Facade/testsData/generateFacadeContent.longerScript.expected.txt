
.INCLUDE "${ORIGINAL_TEMPLATE_DIR}/_includes/_header.tpl"
FRAMEWORK_ROOT_DIR="$(cd "${CURRENT_DIR}/.." && pwd -P)"
FRAMEWORK_SRC_DIR="${FRAMEWORK_ROOT_DIR}/src"
.INCLUDE "${ORIGINAL_TEMPLATE_DIR}/_includes/_load.tpl"

HELP="$(
  cat <<EOF
${__HELP_TITLE}Description:${__HELP_NORMAL} build files using build.sh
and check if bin file has been updated, if yes return exit code > 0

${__HELP_TITLE}Usage:${__HELP_NORMAL} ${SCRIPT_NAME} [--ignore-missing]

    --ignore-missing  do not exit with error for missing files
                      useful when committing because were not existing before

INTERNAL TOOL

.INCLUDE "$(dynamicTemplateDir _includes/author.tpl)"
EOF
)"
